src/coreclr/vm/threadsuspend.cpp

```diff
@@ -5917,7 +5945,6 @@ void HandleGCSuspensionForInterruptedThread(CONTEXT *interruptedContext)
         // If the thread is at a GC safe point, push a RedirectedThreadFrame with
         // the interrupted context and pulse the GC mode so that GC can proceed.
         FrameWithCookie<RedirectedThreadFrame> frame(interruptedContext);
-        pThread->SetSavedRedirectContext(NULL);
```

Comment:
Don't we need to mark the redirect context as in use here?

Reply:
No, in this case `interruptedContext` is stack-allocated in the signal handler. It is not the redirect instance, which we are guarding for under/over use. Marking "in use" here will detect a different instance and trigger an assert.

The original `SetSavedRedirectContext(NULL)` was a bit suspicious since it was not balanced with a "returning" set. If we ever had an actual redirect instance allocated, it might have leaked here.